contrib/OpenInExplorer.lua

```diff
@@ -0,0 +1,69 @@
+--[[OpenInExplorer plugin for darktable
+
+  copyright (c) 2018  Kevin Ertel
+  
+  darktable is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+  
+  darktable is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+  
+  You should have received a copy of the GNU General Public License
+  along with darktable.  If not, see <http://www.gnu.org/licenses/>.
+]]
+
+--[[About this plugin
+This plugin adds the module "OpenInExplorer" to darktable's lighttable view
+
+----REQUIRED SOFTWARE----
+Microsoft Windows Operating System
+
+----USAGE----
+Install: (see here for more detail: https://github.com/darktable-org/lua-scripts )
+ 1) Copy this file in to your "lua/contrib" folder where all other scripts reside. 
+ 2) Require this file in your luarc file, as with any other dt plug-in
+
+Select the photo(s) you wish to find in explorer and press "Go to Folder". 
+A file explorer window will be opened for each selected file at the file's location; the file will be highlighted.
+
+----KNOWN ISSUES----
+]]
+
+local dt = require "darktable"
+local dsys = require "lib/dtutils.system"
+
+--Detect OS and modify accordingly--	
+local proper_install = false
+if dt.configuration.running_os == "windows" then
+	proper_install = true
+end
+
+-- FUNCTION --
+local function OpenInExplorer() --Open in Explorer
+	--Inits--
+	if not proper_install then
+		return
+	end
+	local images = dt.gui.selection()
+	local curr_image = ""
+	
+	for _,image in pairs(images) do 
+		curr_image = image.path..'\\'..image.filename
+		local run_cmd = "explorer.exe /select, "..curr_image

```

Comment:
macos is like windows in that there is one file manager.  You can open a directory with _open /path/to/directory/_ or open with an image selected with _open -R /path/to/image.xxx_.  

Nautilus is the default installed file manager on linux.  You could start with this.  With nautilus you can open the directory and select the file with _nautilus file:///path/to/image.xxx_  With nautilus you can have multiple arguments and a file manager window will open with each one.

I've figured out a method to determine the default file manager in linux, but I have to build a VM and test it.  I didn't want to clutter my install with multiple file managers.

Reply:
I think it would make sense to eventually make this cross platform compatible as you suggest. I might not do it immediately. but this does make sense to do.